### DISCUSSION TITLE
Force Refresh During Debugging

### DISCUSSION THREAD

**Discussion by SPL21GL** [AvaloniaUI/Avalonia#14574] [Q&A] [answered]
I have made a custom Turtle-Graphics-Control in Avalonia on Linux, everything works perfect as expected :)
Here's the code: https://github.com/HTL-Bulme/AvaloniaTurtle

![turtle](https://github.com/AvaloniaUI/Avalonia/assets/90681367/47cae544-9388-4c04-8c1f-946bd9cc5c2a)

But when I enter Debug Mode and step through the App **Step-By-Step** in Rider, the UI **does not refresh**.
What I have found out, is that the render Method of the Custom control is not called until the Step-By-Step Execution is left.

I have tried calling: `InvalidateVisual()` and i tried starting the UI in a different thread. No success so far.

Is there a way to "Wake-Up" the UI thread in debug mode to force refresh the UI?

**Comment by kekekeks** [comment]
UI update happens **after** Render methods for every single control gets updated.

Use something like:
```
async void StepByStep()
{
            foreach (var command in commands)
            {
                command.Execute(state, context);
                InvalidateVisual();
                await Task.Delay(50);
            }
}
```

Render should be rendering only the current frame

**Reply by SPL21GL** [reply, marked as answer]
Indeed `Task.Delay(500).Wait()` did solve the Problem. Thanks
